flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/source/meta/split/StreamSplit.java

```diff
@@ -159,7 +159,15 @@ public String toString() {
     // -------------------------------------------------------------------
     public static StreamSplit appendFinishedSplitInfos(
             StreamSplit streamSplit, List<FinishedSnapshotSplitInfo> splitInfos) {
+        // re-calculate the starting changelog offset after the new table added
+        Offset startingOffset = streamSplit.getStartingOffset();
+        for (FinishedSnapshotSplitInfo splitInfo : splitInfos) {
+            if (splitInfo.getHighWatermark().isBefore(startingOffset)) {
+                startingOffset = splitInfo.getHighWatermark();
+            }
+        }
         splitInfos.addAll(streamSplit.getFinishedSnapshotSplitInfos());
+
         return new StreamSplit(
                 streamSplit.splitId,
                 streamSplit.getStartingOffset(),
```

Comment:
CMIIW, but seems newly added code just calculated the earliest starting offset into `startingOffset` but didn't really use it to generate new `StreamSplit`. Maybe missed a change here?

```suggestion
        return new StreamSplit(
                streamSplit.splitId,
                startingOffset,
```

Reply:
It seems true.

Reply:
done it